desktop/cmp/rest/impl/RestForm.js

```diff
@@ -7,106 +7,90 @@
 
 import {Component} from 'react';
 import {dialog, dialogBody} from '@xh/hoist/kit/blueprint';
-import {HoistComponent, elemFactory, XH} from '@xh/hoist/core';
+import {HoistComponent, elemFactory} from '@xh/hoist/core';
 import {mask} from '@xh/hoist/desktop/cmp/mask';
 import {toolbar} from '@xh/hoist/desktop/cmp/toolbar';
 import {filler, vframe} from '@xh/hoist/cmp/layout';
 import {button} from '@xh/hoist/desktop/cmp/button';
+import {form} from '@xh/hoist/cmp/form';
 import {Icon} from '@xh/hoist/icon';
 import {recordActionBar} from '@xh/hoist/desktop/cmp/record';
 
-import {restControl} from './RestControl';
-
 import './RestForm.scss';
-import {RestFormModel} from './RestFormModel';
+import {restFormField} from './RestFormField';
 
 @HoistComponent
 export class RestForm extends Component {
 
-    static modelClass = RestFormModel;
-
     baseClassName = 'xh-rest-form';
 
     render() {
-        const {record, isAdd, isWritable} = this.model;
-        if (!record) return null;
+        const {model} = this,
+            {isAdd, readonly, isOpen} = this.model;
+        if (!isOpen) return null;
 
         return dialog({
-            title: isAdd ? 'Add Record' : (isWritable ? 'Edit Record' : 'View Record'),
+            title: isAdd ? 'Add Record' : (!readonly ? 'Edit Record' : 'View Record'),
             icon: isAdd ? Icon.add() : Icon.edit(),
             className: this.getClassName(),
             isOpen: true,
             isCloseButtonShown: false,
-            items: this.getDialogItems(),
+            items: [
+                this.renderForm(),
+                this.renderToolbar(),
+                mask({model: model.loadModel, spinner: true})
+            ],
             transitionName: 'none'
         });
     }
 
     //------------------------
     // Implementation
     //------------------------
-    getDialogItems() {
-        const model = this.model;
-        return [
-            dialogBody(this.getForm()),
-            toolbar(this.getButtons()),
-            mask({model: model.loadModel, spinner: true})
-        ];
-    }
+    renderForm() {
+        const {model} = this,
+            formFields = model.editors.map(editor => {
+                return restFormField({editor, model});
+            });
 
-    getForm() {
-        const {isWritable} = this.model;
-        return vframe(
-            this.model.controlModels.map((model,
-                idx) => restControl({model, disabled: !isWritable, autoFocus: idx === 0}))
+        return dialogBody(
+            form({
+                model: model.formModel,
+                fieldDefaults: {
+                    commitOnChange: true,
```

Comment:
Makes me again think we should consider flipping the default here.  This seems like a recurring pattern where we set to true to avoid bugs around edits not flushing as quickly as expected to update validation, etc.  

We could still keep the feature, but make it opt-in for controls where not waiting to commit will thrash a back-end search-as-you-type, etc. Those do seem like important use cases, but also perhaps the exception vs. the rule.